src/daft-csv/src/read.rs

```diff
@@ -137,24 +137,26 @@ pub fn read_csv_bulk(
 
 #[allow(clippy::too_many_arguments)]
 pub async fn stream_csv(
-    uri: &str,
+    uri: String,
     convert_options: Option<CsvConvertOptions>,
     parse_options: Option<CsvParseOptions>,
     read_options: Option<CsvReadOptions>,
     io_client: Arc<IOClient>,
     io_stats: Option<IOStatsRef>,
     max_chunks_in_flight: Option<usize>,
-) -> DaftResult<impl Stream<Item = DaftResult<Vec<Table>>> + Send> {
-    stream_csv_single(
-        uri,
+) -> DaftResult<BoxStream<'static, DaftResult<Vec<Table>>>> {
```

Comment:
Switched to `BoxStream` because we need to return different stream implementations based on runtime conditions. `impl Stream` can't handle multiple concrete types, but `BoxStream` allows this flexibility through dynamic dispatch.